### PR TITLE
Make UpdateMemberForm Responsive

### PR DESCRIPTION
**Pull request by AnukratiMehta**
# Closes issue #155 

## Description

This pull request addresses the offscreen form issue on small screens when editing user information on the profile page. Additionally, it enhances the `Labeled` component to support full-width rendering based on the `fullWidth` prop.

#### Form Component Changes:
- Added `max-w-full` and increased padding (`p-5`) to the form container for better responsiveness.
- Adjusted the flex properties for better wrapping and responsiveness on both form sections.
- Modified the width of the select input and text input to be responsive and controlled by media queries.
- Added a new style block with a media query to set a maximum width for the `.w-xs` class on screens larger than 768px.

#### Labeled Component Changes:
- Added a new prop `fullWidth` to the `Labeled` component, defaulting to `false`.
- Conditionally applied the `w-full` and `md:w-auto` classes based on the value of the `fullWidth` prop, allowing the component to adapt to different layout requirements.

## Screenshots

lg
md
sm

## Checklist

- [x] Tested the changes on different screen sizes.
- [x] No new linting errors or warnings.

### CODE REVIEW

File: src/routes/members/[studentId]/UpdateMemberForm.svelte [hunk 1]
```diff
@@ -46,12 +46,17 @@
...
     />
   </div>
-  <div class="flex gap-2 [&>*:nth-child(3)]:flex-1">
-    <Labeled label="Program" id="classProgramme" error={$errors.classProgramme}>
+  <div class="flex w-full flex-wrap gap-2 [&>*:nth-child(3)]:flex-1">
+    <Labeled
+      label="Program"
+      id="classProgramme"
+      error={$errors.classProgramme}
+      fullWidth={true}
```
Comment (danieladugyan) [inline src/routes/members/[studentId]/UpdateMemberForm.svelte:54]: This can be simplified:
```suggestion
      fullWidth
```

File: src/routes/members/[studentId]/UpdateMemberForm.svelte [hunk 2]
```diff
@@ -60,12 +65,17 @@
...
+      id="classProgramme"
+      error={$errors.classYear}
+      fullWidth={true}
+    >
       <input
         type="number"
         name="classYear"
         id="classYear"
-        class="input input-bordered"
+        class="w-xs input input-bordered"
```
Comment (danieladugyan) [inline src/routes/members/[studentId]/UpdateMemberForm.svelte:78]: `w-xs` can be replaced by a Tailwind class:
```suggestion
        class="input input-bordered md:max-w-20"
```

File: src/routes/members/[studentId]/UpdateMemberForm.svelte [hunk 3]
```diff
@@ -46,12 +46,17 @@
...
+      label="Program"
+      id="classProgramme"
+      error={$errors.classProgramme}
+      fullWidth={true}
+    >
       <select
         id="classProgramme"
         name="classProgramme"
-        class="select select-bordered w-full max-w-xs"
+        class="w-xs select select-bordered w-full"
```
Comment (danieladugyan) [inline src/routes/members/[studentId]/UpdateMemberForm.svelte:59]: `w-xs` can be replaced by a Tailwind class:
```suggestion
        class="select select-bordered w-full md:max-w-20"
```

File: src/routes/members/[studentId]/UpdateMemberForm.svelte [hunk 4]
```diff
@@ -80,3 +90,11 @@
...
   <button type="submit" class="btn btn-secondary mt-4">Spara</button>
 </form>
+
+<style>
+  @media (min-width: 768px) {
+    .w-xs {
+      max-width: 20rem;
+    }
+  }
+</style>
```
Comment (danieladugyan) [inline src/routes/members/[studentId]/UpdateMemberForm.svelte:100]:
```suggestion
```

File: src/lib/components/Labeled.svelte [hunk 5]
```diff
@@ -3,9 +3,14 @@
...
   export let error: string | string[] | undefined = undefined;
+  export let fullWidth: boolean = false;
 </script>
 
-<div class="relative inline-flex flex-col items-stretch">
+<div
+  class={`relative inline-flex flex-col items-stretch ${
+    fullWidth ? "w-full md:w-auto" : ""
+  }`}
+>
```
Comment (danieladugyan) [inline src/lib/components/Labeled.svelte:13]: Is this change needed? When I tested your changes it looked just as good without it.

File: src/routes/members/[studentId]/UpdateMemberForm.svelte [hunk 6]
```diff
@@ -21,9 +21,9 @@
   method="POST"
   action="?/update"
   use:enhance
-  class="form-control gap-2"
+  class="form-control max-w-full gap-2 p-5"
```
Comment (danieladugyan) [inline src/routes/members/[studentId]/UpdateMemberForm.svelte:24]: The padding isn't needed here, since the `UpdateMemberForm` component is used inside a grid that has a gap between its columns. This gap already provides enough whitespace.
```suggestion
  class="form-control max-w-full gap-2"
```

**Review by danieladugyan — CHANGES_REQUESTED** [review]
Thanks for your PR! Other than some minor details in the code it looks good.

File: src/routes/members/[studentId]/UpdateMemberForm.svelte (hunk 2 above)
Comment (AnukratiMehta) [reply src/routes/members/[studentId]/UpdateMemberForm.svelte:78]: Yes, that's what I wanted to do but was met with this error: ```Attributes should not contain ':' characters to prevent ambiguity with Svelte directives(illegal-attribute-character) ```

File: src/routes/members/[studentId]/UpdateMemberForm.svelte (hunk 3 above)
Comment (AnukratiMehta) [reply src/routes/members/[studentId]/UpdateMemberForm.svelte:59]: Yes, that's what I wanted to do but was met with this error: ```Attributes should not contain ':' characters to prevent ambiguity with Svelte directives(illegal-attribute-character) ```

File: src/lib/components/Labeled.svelte (hunk 5 above)
Comment (AnukratiMehta) [reply src/lib/components/Labeled.svelte:13]: On small screens, the Program and Year fields looked odd to me because the rest of the elements take the full width of the screen. But it's up to your discretion and I'm happy to remove it if you think it looks good.

File: src/routes/members/[studentId]/UpdateMemberForm.svelte (hunk 2 above)
Comment (danieladugyan) [reply src/routes/members/[studentId]/UpdateMemberForm.svelte:78]: That's odd, it should definitely work.

File: src/routes/members/[studentId]/UpdateMemberForm.svelte [hunk 7]
```diff
@@ -60,12 +65,17 @@
         {/each}
       </select>
     </Labeled>
-    <Labeled label="Year" id="classProgramme" error={$errors.classYear}>
+    <Labeled
+      label="Year"
+      id="classProgramme"
+      error={$errors.classYear}
+      fullWidth={true}
```
Comment (danieladugyan) [inline src/routes/members/[studentId]/UpdateMemberForm.svelte:72]:
```suggestion
      fullWidth
```

File: src/lib/components/Labeled.svelte (hunk 5 above)
Comment (danieladugyan) [reply src/lib/components/Labeled.svelte:13]: You're right, I must have missed that!

**Review by danieladugyan — CHANGES_REQUESTED** [review]

File: src/routes/members/[studentId]/UpdateMemberForm.svelte (hunk 2 above)
Comment (AnukratiMehta) [reply src/routes/members/[studentId]/UpdateMemberForm.svelte:78]: In that case, should I simply commit your suggestions?

**Comment by danieladugyan** [comment]
Everything looks great now! But you need to run `pnpm format:fix` to fix some minor code style issues to make the CI pipeline pass.
